subsys/bluetooth/audio/Kconfig.has

```diff
@@ -7,6 +7,8 @@
 menuconfig BT_HAS
 	bool "Hearing Access Service support [EXPERIMENTAL]"
 	select EXPERIMENTAL
+	select BT_SMP_SC_PAIR_ONLY
+	depends on BT_SMP_MIN_ENC_KEY_SIZE = 16
```

Comment:
The final question:

Should we use `select BT_SMP_SC_PAIR_ONLY` or `depends on BT_SMP_SC_PAIR_ONLY` where the former is going to be easier to use, but is inconsistent with `depends on BT_SMP_MIN_ENC_KEY_SIZE = 16`.


Previously we've in LE Audio have received significant pushback on using `depends on` when adding dependencies, because it hides Kconfig options in the `menuconfig` GUI, but it's hard to avoid for something like `depends on BT_SMP_MIN_ENC_KEY_SIZE = 16`. 

The alternative we were (forced) to adopt by a reviewer with regards to TX buffer count, were to add a 

`default 16 if BT_HAS` in `BT_SMP_MIN_ENC_KEY_SIZE` instead, and then add a `BUILD_ASSERT`, to actually ensure the value was set... 

Personally I like `depends on` a lot better, but I also do not use the GUI. 

@asbjornsabo @jhedberg @MariuszSkamra  What are you thoughts on this?

Reply:
`depends on` is usually the safer option. Btw, I recommend reading this section in the Zephyr documentation if you haven't already: https://docs.zephyrproject.org/latest/guides/build/kconfig/tips.html#select-pitfalls

Reply:
> The final question:
> 
> Should we use `select BT_SMP_SC_PAIR_ONLY` or `depends on BT_SMP_SC_PAIR_ONLY` where the former is going to be easier to use, but is inconsistent with `depends on BT_SMP_MIN_ENC_KEY_SIZE = 16`.

@Thalley I am not pedantic. Does it really matter if one dependency is defined with `select` and another `depends on` ?
Both are used together in many places across the Zephyr.

> `depends on` is usually the safer option. Btw, I recommend reading this section in the Zephyr documentation if you haven't already: https://docs.zephyrproject.org/latest/guides/build/kconfig/tips.html#select-pitfalls

@jhedberg Thanks for the link, I'm aware that `select` does not resolve dependencies, but I saw it's already used in few places to enable `BT_SMP_SC_PAIR_ONLY`.

I don't have a opinion regarding this. If you say I should use `depends on` I'll change it.

Reply:
Either way ensures that the Kconfig is enabled :)